File: LogAndroid/TrustarApp/src/main/kotlin/jp/co/toukei/log/trustar/rest/post/CollectionPost.kt
package jp.co.toukei.log.trustar.rest.post

import jp.co.toukei.log.lib.util.RequestBodyJson
import jp.co.toukei.log.trustar.db.result.entity.CommonCollectionResult
import jp.co.toukei.log.trustar.forEachOptObject
import jp.co.toukei.log.trustar.user.UserInfo
import org.json.JSONArray
import org.json.JSONObject
import third.jsonObj

class CollectionPost(
        private val userInfo: UserInfo?,
        @JvmField val collections: List<CommonCollectionResult>
) : RequestBodyJson {

    override fun jsonBody(): JSONObject {
        return jsonObj {
            "collectionResults" v JSONArray().put(
                    jsonObj {
                        "companyCd" v userInfo?.companyCd
                        "userId" v userInfo?.userId
                        "addCollection" v JSONArray().also { t ->
                            collections.forEach { a ->
                                a.postJson.forEachOptObject { _, r ->
                                    if (r != null) t.put(r)
                                }
                            }
                        }
                    }
            )
        }
    }
}

File: LogiOS/TrustarApp/Services/Request/CollectionPost.swift
//
//  CollectionPost.swift
//  TrustarApp
//
//  Created by CuongNguyen on 15/03/2022.
//  Copyright © 2022 DionSoftware. All rights reserved.
//

import Foundation

class CollectionPost {
    var userInfo: UserInfo?
    var collections: [CommonCollectionResult]
    
    init(userInfo: UserInfo?,
         collections: [CommonCollectionResult]) {
        self.userInfo = userInfo
        self.collections = collections
    }
    
    class ResultClass: Codable {
        var collectionResults: [CollectionResult] = []
    }
    
    class CollectionResult: Codable {
        var companyCd: String = ""
        var userId: String = ""
        var addCollection: [CommonCollectionResult.CommonRow] = []
    }
    
    func result() -> ResultClass {
        let rs: ResultClass = ResultClass()
        let collectionResult = CollectionResult()
        collectionResult.companyCd = userInfo!.companyCd
        collectionResult.userId = userInfo!.userId
        
        collectionResult.addCollection = []
        for it in collections {
            for r in it.postJson {
                collectionResult.addCollection.append(r)
            }
        }
        
        rs.collectionResults.append(collectionResult)
        return rs
    }
}
